proposed/2022/globalpackagereference.md

```diff
@@ -0,0 +1,125 @@
+
+```xml
+<Project>
+  <ItemGroup>
+    <GlobalPackageReference Include="Nerdbank.GitVersioning" Version="1.0.0" />
+  </ItemGroup>
+</Project>
+```
+
+### Technical explanation
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+`<GlobalPackageReference />` will just be syntactical sugar for a `<PackageReference />` with the following metadata:
+
+* IncludeAssets = `Analyzers;Build;BuildMultitargeting;BuildTransitive`
+* PrivateAssets = `All`
+
+This ensures that when these repository-wide packages are consumed, they don't flow to downstream dependencies and only include assets related to
+build.
+
+This will be achieved using an MSBuild [ItemDefinitionGroup](https://docs.microsoft.com/en-us/visualstudio/msbuild/item-definitions):
+```xml
+  <ItemDefinitionGroup Condition="'$(ManagePackageVersionsCentrally)' == 'true' And '$(RestoreEnableGlobalPackageReference)' != 'false'">
+    <GlobalPackageReference>
+      <!--
+        Default global package references to only consume the Analyzers and Build logic in a package.
+        This helps ensure that the package assets are not passed to the compiler or copied to the
+        output directory.  Having a compile-time reference to a package for all projects in a tree
+        is not recommended.  You should only have "global" references to packages that are used for
+        build.
+      -->
+      <IncludeAssets Condition="'$(RestoreEnableGlobalPackageReferenceIncludeAssetsBuildAnalyzers)' != 'false'">Analyzers;Build;BuildMultitargeting;BuildTransitive</IncludeAssets>
+      <!--
+        Default global package references to have all assets private.  This is because global package
+        references are generally stuff like versioning, signing, etc and should not flow to downstream
+        dependencies.  Also, global package references are already referenced by every project in the
+        tree so we don't need them to be transitive.
+      -->
+      <PrivateAssets Condition="'$(RestoreEnableGlobalPackageReferencePrivateAssetsAll)' != 'false'">All</PrivateAssets>
+    </GlobalPackageReference>
+  </ItemDefinitionGroup>
+```
+
+The following MSBuild properties can be set to change the functionality:
+
+| Property Name | Function | Default value|
+|---|---|---|
+| `ManagePackageVersionsCentrally` | Enables or disables central package management and all dependent features | `false` |
+| `RestoreEnableGlobalPackageReference` | Enables or disables just the concept of `GlobalPackageReference` | `true` |
+| `RestoreEnableGlobalPackageReferenceIncludeAssetsBuildAnalyzers` | Enables or disables defaulting the `IncludeAssets` metadata for global package references | `true` |
+| `RestoreEnableGlobalPackageReferencePrivateAssetsAll` | Enables or disables defaulting the `PrivateAssets` metadata for global package references | `true`|
```

Comment:
Also, instead of opting-in, just as I mentioned detecting `Directory.Packages.props` to enable central package versions, do the same for "Global" package references. We could do this either by detecting "packages" props or by detecting "build" props and non-empty **GPR** items check.